construct/core.py

```diff
@@ -524,10 +524,12 @@ def _parse(self, stream, context):
         except ConstructError:
             if c < self.mincount:
                 raise RangeError("expected %d to %d, found %d" %
-                    (self.mincount, self.maxcout, c), sys.exc_info()[1])
+                    (self.mincount, self.maxcout, c))
             stream.seek(pos)
         return obj
     def _build(self, obj, stream, context):
+        if (not hasattr(obj, '__len__')) or hasattr(obj, 'get'):
```

Comment:
The check is to see if a dictionary was passed instead of a sequence. Without that, you get other weird errors because it passes the keys to subcons to build.